auth/service.cc

```diff
@@ -189,9 +194,13 @@ future<> service::create_keyspace_if_missing(::service::migration_manager& mm) c
 }
 
 future<> service::start(::service::migration_manager& mm) {
-    return once_among_shards([this, &mm] {
-        return create_keyspace_if_missing(mm);
-    }).then([this] {
+    auto create_keyspace_if_missing_or_noop = _maintenance_port
+        ? make_ready_future<>()
+        : once_among_shards([this, &mm] {
+            return create_keyspace_if_missing(mm);
+        });
```

Comment:
I don't understand why we need to disable this when the maintenance port is enabled. This seems like a layering violation to me.

Reply:
From what I understand: all queries go through cql controller which underneath has a reference to auth service.

We want auth to be disabled for maintenance port.

So the maintenance-port instance of cql controller has a reference to a "dummy" auth service that does not create a keyspace and so on. (It cannot, because at the moment of opening maintenance port, we don't have access to the cluster; and the maintenance port is supposed to also be used in situations where we won't have access to the cluster later, so we cannot do auth over distributed tables).

The "old" auth service will still do all of this stuff.

Reply:
@margdoc I think what's confusing is the name `maintenance_port_enabled`.

It's not intuitive that there will be two instances of auth service where one has `maintenance_port_enabled = false` and another has `maintenance_port_enabled = true`. The port cannot be both enabled and disabled at the same time.

The name should instead somehow signify that this instance of the service is "associated to" the maintenance port. Maybe `used_by_maintenance_port`?

Reply:
It is disabled only for maintenance_auth_service which doesn't need that and it can't be created before joining group0.
When regular auth_service is started, it is called as before.

Reply:
Maybe we should invent a better name for this "maintenance_port" parameter.